### ISSUE TITLE
Attract Mode crashes with some videos

### ISSUE THREAD

**Issue by kamzata** [substring/os#29]
Using Attract Mode with some videos ([like this](https://www.progettosnaps.net/videosnaps/mp4/outzone.mp4)) make it crash.

/home/arcade/shared/logs/attract.log
```
Attract-Mode Plus v2.6.2 (Build 0) (Linux, SFML 2.6.0 +FontConfig +7z +Curl)
avcodec 60.3.100 / avformat 60.3.100 / swscale 7.1.100 / avutil 58.2.100 / swresample 4.10.100

Config: /home/arcade/shared/frontends/attract/attract.cfg

*** Initializing display: 'MAME'
 - Loaded master romlist 'MAME' in 51 ms (14117 entries kept, 0 discarded)
 - Constructed 2 filters in 1 ms (28234 comparisons)
Failed to get output info for.primary monitor position
 - Loaded layout: /home/arcade/shared/frontends/attract/layouts/Reflect/ (layout.nut)
munmap_chunk(): invalid pointer
```

Any solutions?

**Comment by substring** [comment]
Hi,

Have you updated recently, or still on a vanilla install ? Are you using X or KMS ? Have you tried a different layout ?

**Comment by kamzata** [comment]
> Hi,
> 
> Have you updated recently, or still on a vanilla install ? Are you using X or KMS ? Have you tried a different layout ?

Once installed, I updated the whole system to the latest packages. I tried both X and KMS, and different layouts: same result.

**Comment by kamzata** [comment]
> Have you tried a different layout ? When have you updated ? Because there are updates "almost" everyday (but don't update yet if you have installed RA)

Yes, I tried different layouts. I installed it around a week ago, I updated the whole system the same day and then I updated yesterday. I have noticed this problem just yesterday since I haven't used videos before.

**Comment by kamzata** [comment]
> Have you tried a different layout ? When have you updated ? Because there are updates "almost" everyday (but don't update yet if you have installed RA)

https://share.icloud.com/photos/010osFL9zw0TnZSOqqjvaFowg

**Comment by kamzata** [comment]
> Have you tried a different layout ? When have you updated ? Because there are updates "almost" everyday (but don't update yet if you have installed RA)

I just installed FFmpeg package but nothing's changed.

**Comment by substring** [comment]
ffmpeg was already installed. I think it's a problem with either the ffmpeg lib, either some trouble in AM. Which is beyond my scope. Does it always happen on the same video ?

**Comment by kamzata** [comment]
> ffmpeg was already installed. I think it's a problem with either the ffmpeg lib, either some trouble in AM. Which is beyond my scope. Does it always happen on the same video ?

Sorry! You're right. I saw the pacman log and ffmpeg was already installed. Why did the attract log change after reinstall ffmpeg?

This is all `pacman.log` entries for ffmpeg since I installed Groovyarcade:

```
[2023-03-16T18:28:46+0000] [ALPM] upgraded ffmpeg (2:5.1.2-1 -> 2:6.0-3)
[2023-03-22T16:30:10+0100] [ALPM] upgraded ffmpeg (2:6.0-3 -> 2:6.0-4)
[2023-03-23T16:47:19+0100] [PACMAN] Running 'pacman -S ffmpeg'
[2023-03-23T16:47:21+0100] [ALPM] transaction started
[2023-03-23T16:47:21+0100] [ALPM] reinstalled ffmpeg (2:6.0-4)
[2023-03-23T16:47:21+0100] [ALPM] transaction completed
[2023-03-23T16:47:21+0100] [ALPM] running '30-systemd-update.hook'...
```

**Comment by substring** [comment]
you can't downgrade ffmpeg since AM+ was compiled against ffmpeg 6, so libs versions won't match and AM won't start.

if it happens with specific videos, the workaround for now would be to reencode them. I can't do much for AM code itself, even less regarding video decoding.

**Comment by kamzata** [comment]
> you can't downgrade ffmpeg since AM+ was compiled against ffmpeg 6, so libs versions won't match and AM won't start.
> 
> if it happens with specific videos, the workaround for now would be to reencode them. I can't do much for AM code itself, even less regarding video decoding.

What about reinstalling Groovyarcade without update? Should it work?

**Comment by substring** [comment]
can't tell. Doing so you'd rollback to ffmpeg5. I have no idea if it would help or not. I just know there was no code change for AM to switch to ffmpeg 6 (unlike ffmpeg 4 -> 5)

**Comment by kamzata** [comment]
I checked the video file codecs. On the left `outrun.mp4` (which seems to work without issues), on the right `outzone.mp4` (which makes Attract Mode crashes every time it play). The codecs seem the same... so... I'm more confused than before...

![outrun-outzone-codecs](https://user-images.githubusercontent.com/1103317/227735714-d54efb04-f401-4c92-9112-ac0dc6ddd34f.png)
